src/main/java/com/celements/web/plugin/cmd/PageLayoutCommand.java

```diff
@@ -19,314 +19,220 @@
-import com.celements.cells.RenderingEngine;
-import com.celements.inheritor.InheritorFactory;
 import com.celements.model.access.IModelAccessFacade;
-import com.celements.model.access.exception.DocumentDeleteException;
 import com.celements.model.access.exception.DocumentNotExistsException;
-import com.celements.model.access.exception.DocumentSaveException;
 import com.celements.model.context.ModelContext;
 import com.celements.model.reference.RefBuilder;
 import com.celements.model.util.ModelUtils;
-import com.celements.model.util.References;
-import com.celements.web.CelConstant;
-import com.google.common.base.Strings;
+import com.celements.pagelayout.LayoutServiceRole;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
 import com.xpn.xwiki.XWikiContext;
 import com.xpn.xwiki.XWikiException;
 import com.xpn.xwiki.doc.XWikiDocument;
 import com.xpn.xwiki.objects.BaseObject;
-import com.xpn.xwiki.plugin.packaging.DocumentInfo;
-import com.xpn.xwiki.plugin.packaging.PackageAPI;
 import com.xpn.xwiki.web.Utils;
 
+/**
+ * @deprecated since 5.3 instead use {@link LayoutServiceRole}
+ */
+@Deprecated
 public class PageLayoutCommand {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(PageLayoutCommand.class);
+
+  @Deprecated
   public static final String SIMPLE_LAYOUT = "SimpleLayout";
 
+  @Deprecated
   public static final String XWIKICFG_CELEMENTS_LAYOUT_DEFAULT = "celements.layout.default";
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(PageLayoutCommand.class);
-
-  QueryManager queryManager;
-
+  @Deprecated
   public static final String PAGE_LAYOUT_PROPERTIES_CLASS_SPACE = "Celements";
+  @Deprecated
   public static final String PAGE_LAYOUT_PROPERTIES_CLASS_DOC = "PageLayoutPropertiesClass";
+  @Deprecated
   public static final String PAGE_LAYOUT_PROPERTIES_CLASS = PAGE_LAYOUT_PROPERTIES_CLASS_SPACE + "."
       + PAGE_LAYOUT_PROPERTIES_CLASS_DOC;
 
+  @Deprecated
   public static final String CEL_LAYOUT_EDITOR_PL_NAME = "CelLayoutEditor";
 
-  private static final String CEL_RENDERING_LAYOUT_CONTEXT_PROPERTY = "celRenderingLayout";
+  private LayoutServiceRole layoutService = Utils.getComponent(LayoutServiceRole.class);
 
-  private static final String CEL_RENDERING_LAYOUT_STACK_PROPERTY = "celRenderingLayoutStack";
-
-  /**
-   * The name of the internal packaging plugin.
-   */
-  private static final String PACKAGEPLUGIN_NAME = "package";
-
-  private InheritorFactory _injectedInheritorFactory;
-
-  private QueryManager getQueryManager() {
-    if (queryManager == null) {
-      queryManager = Utils.getComponent(QueryManager.class);
+  private Map<String, String> convertMap(Map<SpaceReference, String> pageLayoutMap) {
+    Builder<String, String> resultMap = ImmutableMap.<String, String>builder();
+    for (Map.Entry<SpaceReference, String> entry : pageLayoutMap.entrySet()) {
+      resultMap.put(entry.getKey().getName(), entry.getValue());
```

Comment:
can be written more concisely with streams (import needed)
```
return EntryStream.of(pageLayoutMap)
  .mapKeys(SpaceReference::getName)
  .toMap() / .toImmutableMap();
```
also we probably shouldn't return an immutable map of the map was mutable beforehand